typer/main.py

```diff
@@ -936,6 +936,7 @@ def get_click_param(
                 expose_value=parameter_info.expose_value,
                 is_eager=parameter_info.is_eager,
                 envvar=parameter_info.envvar,
+                shell_complete=parameter_info.shell_complete,
```

Comment:
I'm thinking that this one might require the use of `get_param_completion` same as the `autocompletion` on the line below (940). Looking at the code of `get_param_completion` it has some features that are probably applicable here too: mapping the parameters so that not all completion parameters are required (e.g. one can use just the `incomplete: str` instead of needing also the context.

Reply:
Hi Daniel, good catch! I had the same thought, bit if you look just above at line 905 you'll see shell_complete is not wrapped in get_param_completion for Options. Also, the implementation of get_param_completion only works for the autocomplete interface where the second argument is the list of string arguments appearing on the command line before the parameter in question where as in the new shell_complete interface the middle parameter is an instance of click.Parameter.

I do not disagree that it would be nice to wrap shell_complete with get_param_completion like behavior because often the context and Parameter are unnecessary - but this would be a deeper change that would also touch an existing interface and it was my hope to keep this PR as simple as possible to hopefully get it in a 0.9.1 bug fix release. If there's no expectation of that, I'd be happy to submit a chunkier PR for a 0.10 release...

Reply:
Makes sense as you say. I haven't looked at the rest of the code.